Forte.EpiServer.AzureSearch/Events/SearchEventHandler.cs

```diff
@@ -12,92 +13,125 @@ namespace Forte.EpiServer.AzureSearch.Events
+                    foreach (var blockParentPageLink in blockParentPagesLinks)
+                    {
+                        var blockParentPageDocuments =
+                            _pageSearchEventHandler.GetPageContentDocuments(blockParentPageLink);
+                        documentsToIndex.AddRange(blockParentPageDocuments);
+                    }
+                    break;
+            }
+
+            if (documentsToIndex.IsNullOrEmpty() == false)
+            {
+                Task.Run(() => _azureSearchService.IndexAsync(documentsToIndex));
             }
-            
-            var contentInAllLanguageVersions = _contentLoader.GetAllLanguageVersions(contentEventArgs.Content.ContentLink);
-            var documents = contentInAllLanguageVersions.Select(c => _contentDocumentBuilder.Build(c));
-            
-            Task.Run(() => _azureSearchService.IndexAsync(documents));
         }
 
         public void OnMovedContent(object sender, ContentEventArgs contentEventArgs)
         {
-            if (contentEventArgs.TargetLink == ContentReference.WasteBasket)
-            {
-                DeleteTreeFromIndex(contentEventArgs.Content);
-            }
-            else
+            var content = contentEventArgs.Content;
+            switch (content)
             {
-                UpdateTreeInIndex(contentEventArgs.Content);
+                case PageData _:
+                    if (contentEventArgs.TargetLink == ContentReference.WasteBasket)
+                    {
+                        DeletePageTreeFromIndex(content);
+                    }
+                    else
+                    {
+                        UpdatePageTreeInIndex(content);
+                    }
+                    break;
+                case BlockData _:
+                    if (contentEventArgs.TargetLink == ContentReference.WasteBasket)
+                    {
+                        UpdateBlockParentPagesInIndex(content);
+                    }
+                    break;
             }
         }
         
         public void OnSavingContent(object sender, ContentEventArgs contentEventArgs)
         {
             var content = contentEventArgs.Content;
-            var pagePreviousVersion = GetPagePreviousVersion(content.ContentLink);
-            if (pagePreviousVersion != null && IsContentMarkedAsExpired(contentEventArgs, pagePreviousVersion))
+            var contentPreviousVersion = GetContentPreviousVersion(content.ContentLink);
+            if (contentPreviousVersion != null && IsContentMarkedAsExpired(contentEventArgs, contentPreviousVersion))
             {
-                DeleteTreeFromIndex(content);
+                switch (content)
+                {
+                    case PageData _:
+                        DeletePageTreeFromIndex(content);
+                        break;
+                    case BlockData _:
+                        UpdateBlockParentPagesInIndex(content);
```

Comment:
why do we perform delete in case of PageData and update in case of BlockData ?

Reply:
If the page is outdated we want it deleted from the index.
If the block is outdated we want all the block parent pages to reindex, in order to not display block content in their search content results.
That is because we don't index blocks individually, but only as parts of the indexed pages.